sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,297 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
+ * @param[in] low_index An index into the original az_span indicating where the returned az_span
+ * will start
+ * @param[in] high_index An index into the original az_span indicting where the returned az_span
+ * should stop. The byte at the high_index is NOT included in the returned az_span.
+ * @return An az_span (view) into a portion (from low_index to high_index - 1) of the original
+ * az_span.
  */
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high_index);
 
 /**
- * @brief Returns `true` if a content of the @a span is equal to a content of the @b
- * span.
+ * @brief az_span_is_content_equal returns `true` if the lengths and bytes refered by \p span1 and
+ * \p span2 are identical.
+ *
+ * @return Returns true if the lengths of both spans are identical and the bytes in both spans are
+ * also identical.
  */
-AZ_NODISCARD AZ_INLINE bool az_span_is_equal(az_span a, az_span b)
+AZ_NODISCARD AZ_INLINE bool az_span_is_content_equal(az_span span1, az_span span2)
 {
-  return az_span_length(a) == az_span_length(b)
-      && memcmp(az_span_ptr(a), az_span_ptr(b), az_span_length(a)) == 0;
+  return az_span_length(span1) == az_span_length(span2)
+      && memcmp(az_span_ptr(span1), az_span_ptr(span2), az_span_length(span1)) == 0;
 }
 
 /**
- * @brief Returns `true` if a content of the @a span is equal to a content of the @b
- * span using case-insensitive compare.
+ * @brief az_span_is_content_equal_ignoring_case returns `true` if the lengths and characters
+ * refered to by \p span1 and \p span2 are identical except for case. This function assumes the bytes in
+ * both spans are ASCII characters.
  *
- * @param a comparing span
- * @param b comparing span
- * @return AZ_NODISCARD az_span_is_equal_content_ignoring_case
+ * @return Returns true if the lengths of both spans are identical and the ASCII characters in both
+ * spans are also identical except for case.
  */
-AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span a, az_span b);
+AZ_NODISCARD bool az_span_is_content_equal_ignoring_case(az_span span1, az_span span2);
 
 /**
- * @brief convert all the span content to uint64
+ * @brief az_span_to_str copies a source span containing a string (not 0-terminated) to a
+ destination char buffer and appends the 0-terminating byte.
+
+ * The buffer refered to by destination must have a max_size that is at least 1 byte bigger
+ than the source az_span
+ *  the string in converts \p destination span to zero-terminated str. Content is copied to \p source
+ buffer and then \0 is
+ * addeed at the end. Then out_result will be created out of buffer
  *
- * @param self source span
- * @param out result span
- * @return AZ_NODISCARD az_span_to_uint64
+ * @param[in] destination A pointer to a buffer where the string should be copied
+ * @param[in] destination_max_size The maximum size of the buffer refered to by destination
+ * @param[in] source The az_span containing the not-0-terminated string
```

Comment:
```suggestion
 * @param[in] source The az_span containing the non-0-terminated characters.
```

Reply:
There is no such thing as non-0-terminated characters. You can have a non-0-terminated STRING of characters. Or a non-0-terminated string.